### DISCUSSION TITLE
backpack/calendar-operation - bug/inconsistency trying to populate event extendedProps using getCalendarFieldsMap()

### DISCUSSION THREAD

**Discussion by tomsb** [Laravel-Backpack/community-forum#1268] [General]
`extendedProps` typically contains additional event data, but it does not always include all the fields defined in `getCalendarFieldsMap()`. For example, I have the following fields in `getCalendarFieldsMap()`:

```
'module_number' => 'module_number',
'pk' => 'pk',
```
The `module_number` field appeared in `event.extendedProps` (when I clicked on the event), but `pk` did not. I am unsure why `pk` was missing initially. However, after I explicitly added `extendedProps` in the `eventData`, the `pk` field appeared in `extendedProps`.

This behavior is strange. I did not investigate further to understand why this inconsistency occurred.

#### File: calendar_view.blade.php (From line 269)
```diff
 let eventData = {
      id: entry[fieldMap.id],
      title: entry[fieldMap.title],
      start: entry[fieldMap.start],
      end: entry[fieldMap.end],
      color: entry[fieldMap.background_color],
      textColor: entry[fieldMap.text_color],
      allDay: !entry[fieldMap.end],
      editable,
      buttons: entry.buttons,
      ...extra,
+      extendedProps: {
+          ...extra,
+      },
  };

[a comment by an automated account is not shown]

**Comment by jcastroa87** [comment]
Hello @tomsb 

Can you share with me the migration and model to replicate the problem on my side.

Can you please share versions too:
```
php artisan backpack:version
```

Thanks.

Cheers.

**Comment by tomsb** [comment]
Suddenly, I can no longer reproduce this behaviour. I performed some updates and can't easily revert back.

It seems I misunderstood how everything fits together.

Here's what I wanted to achieve: adding a custom value to the `event.extendedProps.` The steps I followed:

1. Use `getCalendarEntries` as `traitGetCalendarEntries` (alias the original trait method)
2. Add custom field like this:
```
public function getCalendarEntries(Request $request): Collection
    {
        $entries = $this->traitGetCalendarEntries($request);
        $entries = $entries->map(function ($entry) {
            $entry['foo'] = 'bar';
```

However, this didn't work as expected. It turns out that I also need to add the same custom field in the `getCalendarFieldsMap()` method for it to appear in `event.extendedProps` (this becomes clear by looking at `calendar_view.blade.php`)

This detail might be worth adding to the calendar documentation. Should we close this issue?
